src/ess/polarization/base.py

```diff
@@ -0,0 +1,200 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
+from typing import NewType, TypeVar
+
+import sciline as sl
+import scipp as sc
+
+Up = NewType('Up', int)
+Down = NewType('Down', int)
+Unpolarized = NewType('Unpolarized', int)
+Spin = TypeVar('Spin', Up, Down, Unpolarized)
+PolarizerSpin = TypeVar('PolarizerSpin', Up, Down)
+AnalyzerSpin = TypeVar('AnalyzerSpin', Up, Down)
+
+Analyzer = NewType('Analyzer', str)
+Polarizer = NewType('Polarizer', str)
+Cell = TypeVar('Cell', Analyzer, Polarizer)
+
```

Comment:
According to my comments below, here should be added:
Cell in = NewType('Cell in', str)
Cell out = NewType('Cell out', str)
Cell position = TypeVar('Cell position',cell in, cell out)

Reply:
I think this is a misunderstanding, `Cell` is either `Polarizer` or `Analyzer` (and base on what you wrote above I am thinking we would need a "neither" case as well).

Reply:
yes, actually the "neither" case would be what I meant with "cell out", which basically is the same reference measurement for both cells (just nothing in the beam). So yes, if we define another "neither" in cell, then the "cell position" is not needed